crate/utils/src/crypto/cover_crypt/attributes.rs

```diff
@@ -239,3 +241,51 @@ pub fn header_uid_from_attributes(attributes: &Attributes) -> Result<&[u8], Kmip
         ))
     }
 }
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum EditPolicyAction {
+    RotateAttributes(Vec<abe_policy::Attribute>),
+    ClearOldRotations(Vec<abe_policy::Attribute>),
+    RemoveAttribute(Vec<abe_policy::Attribute>),
+    DisableAttribute(Vec<abe_policy::Attribute>),
+    AddAttribute(Vec<(abe_policy::Attribute, EncryptionHint)>),
+    RenameAttribute(Vec<(abe_policy::Attribute, String)>),
+}
+
+/// Convert an edit policy action to a vendor attribute
+pub fn edit_policy_action_as_vendor_attribute(
+    action: EditPolicyAction,
+) -> Result<VendorAttribute, KmipError> {
+    Ok(VendorAttribute {
+        vendor_identification: VENDOR_ID_COSMIAN.to_owned(),
+        attribute_name: VENDOR_ATTR_COVER_CRYPT_POLICY_EDIT_ACTION.to_owned(),
+        attribute_value: serde_json::to_vec(&action).map_err(|e| {
+            KmipError::InvalidKmipValue(
+                ErrorReason::Invalid_Attribute_Value,
+                format!("failed serializing the CoverCrypt action: {e}"),
+            )
+        })?,
+    })
+}
+
+/// Extract an edit `CoverCrypt` policy action from attributes
+pub fn edit_policy_action_from_attributes(
+    attributes: &Attributes,
+) -> Result<EditPolicyAction, KmipError> {
```

Comment:
Why didn't you take a `VendorAttribute` as argument? This would make it symmetrical with the reverse operation. Plus it seems you are ignoring the error raised here in the caller (`rekey_keypair:85`)... which means that an error to parse the operation (let's say a user misspell) will result in a rotation... 

If you want to keep it here, maybe can you at least differentiate internal errors and default to a rotation only when no vendor attribute is passed?